### PR TITLE
Add Cache-Control to Nginx redirects

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 1]
```diff
@@ -106,6 +93,21 @@ server {
 
     location / {
         try_files $uri /app.php$is_args$query_string;
+
+        add_header Cache-Control "{{ pillar.journal.cache_control }}";
+
+        rewrite '^/content(/elife)?/[0-9]{1,}/e([0-9]{5,})(v.)?(/.*)?$' '{{ public_scheme }}://{{ public_host }}/articles/$2$3' permanent;
```
Comment (thewilkybarkid) [inline salt/journal/config/etc-nginx-sites-available-journal.conf]: Had to move these to isolate the `add_header` in a `location`.

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 2]
```diff
@@ -106,6 +93,21 @@ server {
 
     location / {
         try_files $uri /app.php$is_args$query_string;
+
+        add_header Cache-Control "{{ pillar.journal.cache_control }}";
```
Comment (giorgiosironi) [inline salt/journal/config/etc-nginx-sites-available-journal.conf]: this has a good default but these redirect should maybe warrant a very long expiration time? They are changed only if there's a bug so hours could possibly work?

Comment (thewilkybarkid) [reply salt/journal/config/etc-nginx-sites-available-journal.conf]: These are for legacy paths, and since we can clear easily let's go for a week.

**Comment by giorgiosironi** [comment]
We can clear, but only everything or a single URL without surrogate keys

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 3]
```diff
@@ -135,6 +137,7 @@ server {
     }
 
     location ~ /app_.*.php$ {
+        add_header Cache-Control "{{ pillar.journal.cache_control }}";
```
Comment (thewilkybarkid) [inline salt/journal/config/etc-nginx-sites-available-journal.conf]: This `location` is actually redundant.

**Pull request merged**